tests/test_algorithms.py

```diff
@@ -265,3 +266,34 @@ def test_merge_valid_bitmaps():
     expected = np.array([0x0], dtype=np.uint8)
     result = _merge_valid_bitmaps(a.slice(5, 2), b.slice(3, 2))
     npt.assert_array_equal(result, expected)
+
+
+def test_stringbuilder():
```

Comment:
Please use `parametrize` to have the tests run several times instead of calling it here manually. Also couldn't you add another parameter whether to test the jitted or the non-jitted version of the StringBuilder?

Reply:
Done